app/response_processor.py

```diff
@@ -46,41 +51,34 @@ def service_name(self, url=None):
             elif 'validate' in parts:
                 return 'SDX-VALIDATE'
         except AttributeError as e:
-            self.logger.error(e)
+            self.logger.error(str(e))
 
-    def process(self, encrypted_survey):
-        # decrypt
+    def _process(self, encrypted_survey):
         decrypted_json = self.decrypt_survey(encrypted_survey)
 
         metadata = decrypted_json['metadata']
-        self.logger = self.logger.bind(user_id=metadata['user_id'], ru_ref=metadata['ru_ref'])
+        self.logger = self.logger.bind(user_id=metadata['user_id'],
+                                       ru_ref=metadata['ru_ref'])
 
         if not self.tx_id:
             self.tx_id = decrypted_json.get('tx_id')
         elif self.tx_id != decrypted_json.get('tx_id'):
+            logger.info('tx_ids from decrypted_json and message header do not match.' +
+                        ' Rejecting message',
+                        decrypted_tx_id=decrypted_json.get('tx_id'),
+                        message_tx_id=self.tx_id)
             raise BadMessageError
 
         self.logger = self.logger.bind(tx_id=self.tx_id)
+        self.validate_survey(decrypted_json)
+        self.store_survey(decrypted_json)
 
-        try:
-            self.validate_survey(decrypted_json)
-        except BadMessageError:
-            # If the validation fails, the message is to be marked "invalid"
-            # and then stored. We don't then want to stop processing at this point.
-            decrypted_json['invalid'] = True
-            self.logger.info("Invalid survey data, skipping receipting", tx_id=self.tx_id)
+        if decrypted_json.get("survey_id") != "feedback":
+            self.logger.info("Receipting survey", tx_id=self.tx_id)
```

Comment:
Talking about bound loggers, doesn't it need to be unbound once it has finished dealing with this message?